addons/mrp/wizard/mrp_batch_produce.py

```diff
@@ -85,10 +85,23 @@ def _production_text_to_object(self, mark_done=False):
             components_list.append(components_line)
 
         productions = self.production_id._split_productions({self.production_id: productions_amount})
-        lots = self.env['stock.lot'].create([{
-            'name': lot_name,
-            'product_id': productions.product_id.id
-        } for lot_name in productions_lot_list])
+        lots = self.env['stock.lot'].search(
+            domain=[
+                ('company_id', '=', self.production_id.product_id.company_id.id),
```

Comment:
```suggestion
                ('company_id', 'in', [self.production_id.product_id.company_id.id, False]),
```
Lot can have company_id unset in 17.2